### ISSUE TITLE
Marking events experimental

### ISSUE THREAD

**Issue by ormsbee** [eduNEXT/openedx-events#28]
Someone (not me, but I am completely blanking on who at the moment 😞 ) brought up the idea of marking new events "experimental" for a certain period of time after initial introduction so that we have a little more flexibility to change things before we mark them as "supported" for long term usage. I thought this was a good idea, but I'm curious what others thought.

FYI: @mariajgrimaldi, @felipemontoya, @feanil, @robrap

**Comment by robrap** [comment]
There is a related discussion in https://github.com/eduNEXT/openedx-events/pull/24 about a status of provisional/not-implement vs active/implemented, and how and if to document or discover this.

In terms of the term "experimental", it doesn't sound like you are saying we are experimenting with whether or not the event is needed, but more that you want to try it out in at least a single use case and evolve it as necessary, before having anyone just start using it? Would it be fair to say that _nearly_ all the unimplemented events would be "experimental", and initial implementations would be "experimental" until we decide they are not? Or, did you have a different definition or meaning of "experimental"?

Note: we are also having separate discussions about schema management and evolution as it relates to a use of an event bus. It may be that we'd like the processes to converge, and having a permanent or semi-permanent record of events may force us to improve initial designs, or other potential consequences. This remains to be seen.

**Comment by ormsbee** [comment]
> In terms of the term "experimental", it doesn't sound like you are saying we are experimenting with whether or not the event is needed, but more that you want to try it out in at least a single use case and evolve it as necessary, before having anyone just start using it? Would it be fair to say that nearly all the unimplemented events would be "experimental", and initial implementations would be "experimental" until we decide they are not?

Yes, this exactly. Or maybe a version of it where it becomes official only after the next named release is cut or something. Some intermediate state between "I had this cool idea" and "we're committing to support this for years".

But maybe that's not actually useful. Maybe we might as well start with it at version 1 and quickly deprecate in the following release. I don't have any strong convictions about it, but I wanted to make sure I passed on the thought.
